File: backend/src/main/java/com/erp/service/DashboardService.java
package com.erp.service;

import com.erp.dto.response.AdminDashboardDto;
import com.erp.dto.response.EmployeeDashboardDto;
import com.erp.dto.response.HRDashboardDto;
import com.erp.dto.response.ManagerDashboardDto;

public interface DashboardService {
    AdminDashboardDto getAdminDashboard();
    HRDashboardDto getHRDashboard();
    ManagerDashboardDto getManagerDashboard();
    EmployeeDashboardDto getEmployeeDashboard();
}

File: frontend/src/services/dashboardService.js
import api from './api'

export const dashboardService = {
  getAdminDashboard: async () => {
    const response = await api.get('/api/dashboard/admin')
    return response.data
  },

  getHRDashboard: async () => {
    const response = await api.get('/api/dashboard/hr')
    return response.data
  },

  getManagerDashboard: async () => {
    const response = await api.get('/api/dashboard/manager')
    return response.data
  },

  getEmployeeDashboard: async () => {
    const response = await api.get('/api/dashboard/employee')
    return response.data
  }
}

export default dashboardService
